opencog/reasoning/pln/rules/deduction.scm

```diff
@@ -8,19 +8,9 @@
 (define pln-rule-deduction
     (BindLink
         (VariableList
-            (TypedVariableLink
                 (VariableNode "$A")
-                (TypeNode "ConceptNode")
-            )
-            (TypedVariableLink
                 (VariableNode "$B")
-                (TypeNode "ConceptNode")
-            )
-            (TypedVariableLink
-                (VariableNode "$C")
-                (TypeNode "ConceptNode")
```

Comment:
In the socrates rule, deduction rule is applied on following two atoms:-
```
(InheritanceLink (stv 1.000000 0.991755)
  (ConceptNode "Socrates@46ec3d0f-4535-4d01-87b7-84ef65c25a23")
  (ConceptNode "men@a2905bdd-9214-4717-82c6-dfe21c1263bc")
)
```
and
```
(InheritanceLink (stv 1.000000 1.000000)
  (ConceptNode "men@a2905bdd-9214-4717-82c6-dfe21c1263bc")
  (SatisfyingSetLink (stv 1.000000 1.000000)
    (VariableNode "$X0")
    (EvaluationLink (stv 1.000000 0.000000)
      (PredicateNode "breathe@7f5b37e8-e4b3-4335-a06b-68af470cf354")
      (ListLink (stv 1.000000 0.000000)
        (VariableNode "$X0")
        (ConceptNode "air@9bfe790d-5446-4219-be74-845c0d145fe1")))))
```
Here one of the atom types participating in the deduction rule is SatisfyingSetLink.